Upgradeable - YABTransfer (#79)
yetanotherco/yet-another-bridge: README.md

### Ethereum smart contract

First, make sure you have the necessary dependencies installed in the `contracts/solidity/lib` folder.

> You can install these commands by executing them within the `/contracts/solidity/` folder.

Reviewer: I would rethink this phrase for better clarity. Which commands should i run to install which dependencies?

Author: makes sense, i just updated this

> - **[openzeppelin-contracts-upgradeable]**(https://github.com/OpenZeppelin/openzeppelin-foundry-upgrades)
> `forge install OpenZeppelin/openzeppelin-contracts-upgradeable`
> 
> - **[forge-std]**(https://github.com/foundry-rs/forge-std)
> `forge install foundry-rs/forge-std `

Reviewer: ```suggestion
- [openzeppelin-contracts-upgradeable](https://github.com/OpenZeppelin/openzeppelin-foundry-upgrades)
`forge install OpenZeppelin/openzeppelin-contracts-upgradeable`

- [forge-std](https://github.com/foundry-rs/forge-std)
`forge install foundry-rs/forge-std `
```
i think this .md links are not rendering correctly

Reviewer: I would also separate in different lines the link from the command

Reviewer 2: +1

## Recap

After following this complete README, we should have an ETH smart contract as well as a Starknet smart contract, both connected to act as a bridge between these two chains.

## Upgrade Contracts in Testnet

### Ethereum

> After deploying the `YABTransfer` contract, you can perform upgrades using the `make ethereum-upgrade` command.

Reviewer: ```suggestion
After deploying the `YABTransfer` contract, you can perform upgrades using the following command:
`make ethereum-upgrade`
```
I think commands are more clear and more easily identifiable if they are on their own line

> After deploying the `YABTransfer` contract, you can perform upgrades using the `make ethereum-upgrade` command.
> 
> Please note that executing this command will rebuild `YABTransfer.sol`, deploy the new contract to the network, and utilize Foundry to upgrade the proxy contract by changing its implementation to the newly deployed contract.
> 
> To execute this action, you must be the **owner** of the contract, and you also need to have set the `YAB_TRANSFER_PROXY_ADDRESS` variable in the `.env` file with the proxy's address.

Reviewer: ```suggestion
After deploying the `YABTransfer` contract, you can perform upgrades of the smart contract.

To execute this action, you must be the **owner** of the contract, and you also need to have set the `YAB_TRANSFER_PROXY_ADDRESS` variable in the `.env` file with the proxy's address.
```
The command is stated more clearly and later in this Readme. I think stating commands only once makes the Readme more easily followed step-by-step.

Also, I would move the clarification of:
"Please note that executing this command will rebuild `YABTransfer.sol`, deploy the new contract to the network, and utilize Foundry to upgrade the proxy contract by changing its implementation to the newly deployed contract."
To when the command is stated, around line 261

Author: I think it's right to move the sentence to the line 261 :+1: 
Also, I deleted the NOTE. 

Regarding the other point, I thought that perhaps each section should explicitly state the required variables. Someone might want to go directly to the `upgrade`, and it's important to have all the necessary information in that section.

1. Update the `contracts/solidity/.env` file.

   ```
      ETH_RPC_URL = Infura or Alchemy RPC URL
      ETH_PRIVATE_KEY = private key of your ETH wallet
      ETHERSCAN_API_KEY = API Key to use etherscan to read the Ethereum blockchain
      SN_MESSAGING_ADDRESS = Starknet Messaging address

      # To upgrade Ethereum contract 

>       YAB_TRANSFER_PROXY_ADDRESS=<YAB_TRANSFER_PROXY_ADDRESS> 

Reviewer: i would add this new variable to the first time solidity/.env is named (around line 127).
I think this change + automatic update of YAB_TRANSFER_PROXY_ADDRESS will make the .env config much simpler

Author: done!

>       YAB_TRANSFER_PROXY_ADDRESS=<YAB_TRANSFER_PROXY_ADDRESS> 
>    ```
> 
> **NOTES**:
> - `YAB_TRANSFER_PROXY_ADDRESS` is the proxy for the `YABTransfer` that you want to upgrade.

Reviewer: ```suggestion
      YAB_TRANSFER_PROXY_ADDRESS = The proxy for the `YABTransfer` contract that you want to upgrade.
   ```
```
To follow the format of explaining the .env vars
